R/deps.R

```diff
@@ -107,6 +108,35 @@ compare_versions <- function(a, b) {
   vapply(seq_along(a), function(i) compare_var(a[[i]], b[[i]]), integer(1))
 }
 
+install_dev_remotes <- function(pkg, ...) {
+  pkg <- as.package(pkg)
+
+  if (is.null(pkg$remotes)) {
+    return()
+  }
+
+  types <- lapply(pkg$remotes, dev_remote_type)
+
+  lapply(types, function(type) type$fun(type$repository, ...))
+}
+
+# Parse the remotes field split into pieces and get install_ functions for each
+# remote type
+dev_remote_type <- function(remotes) {
+
+  dev_packages <- trimws(unlist(strsplit(remotes, ",[[:space:]]*")))
```

Comment:
I forgot `trimws()` was only available in new R versions.  We can replace it with `trimws <- function(x, gsub("^[[:space:]]+|[[:space:]]+$", "", x))` so we don't have to bump the version.